### PR TITLE
A lot of devtools-common important fixes

### PR DESCRIPTION
**Pull request by ia3andy**
- Use requireNonNull instead of checkNotNull, and ArrayList as List builder.
- use dir instead of folder in field names
- adding extension was not possible without specifying the version (added test)
- there were a wrong assignment in AppArtifactCoords
- I renamed the devtools-common folder from "common" to "devtools-common"

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/compress/QuarkusProjectCompress.java [hunk 1]
```diff
@@ -17,34 +16,33 @@
...
 
-    @VisibleForTesting
+    // Visible for testing
     static final int DIR_UNIX_MODE = UnixStat.DIR_FLAG | UnixStat.DEFAULT_DIR_PERM;
 
     private QuarkusProjectCompress() {
     }
 
-    public static void zip(final QuarkusProject quarkusProject, final Path targetZipPath, final boolean includeProjectFolder)
+    public static void zip(final Path quarkusProjectFolderPath, final Path targetZipPath, final boolean includeProjectFolder)
```
Comment (aloubyansky) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/compress/QuarkusProjectCompress.java:31]: Folder? Is that a Windows jargon? :) I think all the Java API I've seen uses "directory" or "dir".

Comment (ia3andy) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/compress/QuarkusProjectCompress.java:31]: oh boy, I've used folder at many places 😬
let me rename it all 😅

**Comment by geoand** [comment]
CI doesn't like it one bit :)

**Comment by ia3andy** [comment]
> CI doesn't like it one bit :)

@geoand yeah I forgot to rename the module in the pom.xml :-) I am preparing a new commit

**Comment by ia3andy** [comment]
@gastaldi sorry I found a few bug while doing that PRs, I included them after you review. (I didn't changed the first commits that you reviewed)

**Comment by ia3andy** [comment]
This PR has been tested against code.quarkus.io: https://github.com/quarkusio/code.quarkus.io/commits/compat-with-new-codegen

**Comment by ia3andy** [comment]
@aloubyansky waiting for your go to merge that one.

**Review by aloubyansky — APPROVED** [review]

**Pull request merged**